src/wnaf.rs

```diff
@@ -504,3 +505,28 @@ impl<G: Group, const WINDOW_SIZE: usize> Mul<&WnafScalar<G::Scalar, WINDOW_SIZE>
         wnaf_exp(&self.table, &rhs.wnaf)
     }
 }
+
+#[test]
+fn test_wnaf_form() {
+    fn from_wnaf(wnaf: &Vec<i64>) -> u128 {
+        wnaf.iter().rev().fold(0, |acc, next| {
+            let mut acc = acc * 2;
+            acc += *next as u128;
+            acc
+        })
+    }
+    for w in 2..64 {
+        for e in 0..=u16::MAX {
+            let mut wnaf = vec![];
+            wnaf_form(&mut wnaf, e.to_le_bytes(), w);
+            assert_eq!(e as u128, from_wnaf(&wnaf));
+        }
+    }
+    for w in 2..64 {
+        for e in u128::MAX - 10000..=u128::MAX {
+            let mut wnaf = vec![];
```

Comment:
`wnaf_form` clears the given vector before using it, so this is faster (and also tests that clearing logic as a side-effect):
```suggestion
    }
    let mut wnaf = Vec::with_capacity(129);
    for w in 2..64 {
        for e in 0..=u16::MAX {
            wnaf_form(&mut wnaf, e.to_le_bytes(), w);
            assert_eq!(e as u128, from_wnaf(&wnaf));
        }
    }
    for w in 2..64 {
        for e in u128::MAX - 10000..=u128::MAX {
```